compiler-plugin/src/test/kotlin/arrow/meta/quotes/scope/ObjectDeclarationTest.kt

```diff
@@ -0,0 +1,26 @@
+package arrow.meta.quotes.scope
+
+import arrow.meta.plugin.testing.CompilerTest
+import arrow.meta.plugin.testing.CompilerTest.Companion.source
+import arrow.meta.plugin.testing.assertThis
+import arrow.meta.quotes.scope.plugins.ScopeMetaPlugin
+import org.junit.Test
+
+class ObjectDeclarationTest {
+  
+  private val `object` = """
+                         | //metadebug
+                         | @Deprecated("Test") object Test {
+                         |   fun test() { println("Test") }
+                         |   fun test2() { println("Test2") }
+                         | }""".source
+  
+  @Test
+  fun `validate object declaration scope properties`() {
+    assertThis(CompilerTest(
+      config = { metaDependencies + addMetaPlugins(ScopeMetaPlugin()) },
```

Comment:
so awesome! Thank you for this! I'm going to leave a comment at the top of the review for easy access/docs